src/shared_utilities/TimeVaryingInputs.jl

```diff
@@ -0,0 +1,82 @@
+# TimeVaryingInputs.jl
+#
+# This module contains structs and methods to process external data and evaluate it on the
+# model. This module only concerns with evaluations in time, not in space.
+
+# There are three possible sources of data:
+# 1. Analytic functions that prescribe how a variable has to be set at a given time
+# 2. 1D, single-site data, which is assumed small enough to be saved to memory
+# 3. 2D, global data, which cannot be saved to memory in its entirety.
+#
+# The TimeVaryingInputs module introduces a shared interface for the three cases so that
+# uses and developers do not have to worry about the details of what type of data will be
+# provided. Behind the scenes, we introduce a new type AbstractTimeVaryingInput, that has
+# three concrete implementation, corresponding to the three use cases described above.
+# Constructors will automatically identify which of the three implementations to use based
+# on the input data, and the existence of three concrete structs should be considered an
+# implementation detail.
+#
+# The three TimeVaryingInputs are:
+# - AnalyticTimeVaryingInput,
+# - InterpolatingTimeVaryingInput1D,
+# - InterpolatingTimeVaryingInput3D.
+#
+# Along side these TimeVaryingInputs, we also define InterpolationMethods that implement
```

Comment:
very clear!

Reply:
one more thing to note should be that the time variable needs to be in "simulation time", i.e. seconds since a reference datetime. This is because we evaluate! using `t` from the simulation.

Reply:
Added.